lib/srv/desktop/windows_server_test.go

```diff
@@ -192,3 +195,83 @@ func TestGenerateCredentials(t *testing.T) {
 	require.True(t, foundKeyUsage)
 	require.True(t, foundAltName)
 }
+
+func TestEmitsRecordingEventsOnSend(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+	s := &WindowsService{
+		cfg: WindowsServiceConfig{
+			Clock: clock,
+		},
+	}
+
+	// a fake PNG Frame message
+	encoded := []byte{byte(tdp.TypePNGFrame), 0x01, 0x02}
+
+	ch := make(chan events.AuditEvent)
+	go func() {
+		delay := func() int64 { return 0 }
+		handler := s.makeTDPSendHandler(context.Background(), &channelEmitter{eventsCh: ch}, delay)
+
+		// the handler accepts both the message structure and its encoded form,
+		// but our logic only depends on the encoded form, so pass a nil message
+		var msg tdp.Message = nil
+		handler(msg, encoded)
+	}()
+
+	select {
+	case e := <-ch:
+		dr, ok := e.(*events.DesktopRecording)
+		require.True(t, ok)
+		require.Equal(t, encoded, dr.Message)
+	case <-time.After(1 * time.Second):
```

Comment:
1 second is an eternity for a channel send. I was aiming for a quick failure here rather than a failure that causes CI to to hang until the max timeout. 🤷‍♂️

Reply:
Fair.